File: app/src/main/java/com/product/marcus/nearbygrocery/models/PlaceResponse.java

package com.product.marcus.nearbygrocery.models;

import java.util.List;
import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class PlaceResponse implements Parcelable
{

    @SerializedName("html_attributions")
    @Expose
    private List<Object> htmlAttributions = null;
    @SerializedName("results")
    @Expose
    private List<Result> results = null;
    @SerializedName("status")
    @Expose
    private String status;
    public final static Parcelable.Creator<PlaceResponse> CREATOR = new Creator<PlaceResponse>() {


        @SuppressWarnings({
            "unchecked"
        })
        public PlaceResponse createFromParcel(Parcel in) {
            PlaceResponse instance = new PlaceResponse();
            in.readList(instance.htmlAttributions, (java.lang.Object.class.getClassLoader()));
            in.readList(instance.results, (Result.class.getClassLoader()));
            instance.status = ((String) in.readValue((String.class.getClassLoader())));
            return instance;
        }

        public PlaceResponse[] newArray(int size) {
            return (new PlaceResponse[size]);
        }

    }
    ;

    public List<Object> getHtmlAttributions() {
        return htmlAttributions;
    }

    public void setHtmlAttributions(List<Object> htmlAttributions) {
        this.htmlAttributions = htmlAttributions;
    }

    public List<Result> getResults() {
        return results;
    }

    public void setResults(List<Result> results) {
        this.results = results;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeList(htmlAttributions);
        dest.writeList(results);
        dest.writeValue(status);
    }

    public int describeContents() {
        return  0;
    }

}

File: app/src/main/java/com/product/marcus/nearbygrocery/models/PlaceResponse.kt
package com.product.marcus.nearbygrocery.models

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

class PlaceResponse : Parcelable {

    @SerializedName("html_attributions")
    @Expose
    var htmlAttributions: List<Any>? = null
    @SerializedName("results")
    @Expose
    var results: List<Result>? = null
    @SerializedName("status")
    @Expose
    var status: String? = null

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeList(htmlAttributions)
        dest.writeList(results)
        dest.writeValue(status)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<PlaceResponse> = object : Parcelable.Creator<PlaceResponse> {


            override fun createFromParcel(`in`: Parcel): PlaceResponse {
                val instance = PlaceResponse()
                `in`.readList(instance.htmlAttributions, Any::class.java.classLoader)
                `in`.readList(instance.results, Result::class.java.classLoader)
                instance.status = `in`.readValue(String::class.java.classLoader) as String
                return instance
            }

            override fun newArray(size: Int): Array<PlaceResponse?> {
                return arrayOfNulls(size)
            }

        }
    }

}
